Client/src/Views/UserDatasets/AllUploads.tsx

```diff
@@ -0,0 +1,147 @@
+import React from 'react';
+
+import Icon from 'wdk-client/Components/Icon/IconAlt';
+import { Link } from 'wdk-client/Components';
+import { UserDatasetUpload } from 'wdk-client/Utils/WdkModel';
+import { wrappable } from 'wdk-client/Utils/ComponentUtils';
+import UserDatasetEmptyState from 'wdk-client/Views/UserDatasets/EmptyState';
+
+interface Props {
+  uploadList?: Array<UserDatasetUpload>;
+  errorMessage?: string;
+  actions: {
+    clearMessages: (ids: string[]) => void;
+    cancelCurrentUpload: (id: string) => void;
+  };
+}
+
+
+const ClearAllMessagesButton = (onClickCallback: () => void) => (
+  <button type="submit" className="btn" onClick={onClickCallback}>Clear All Messages</button>
+);
+
+type HeaderProps = {
+  color: string;
+  iconType: string;
+  date: string;
+};
+const UploadHeader = ({color, iconType, date}: HeaderProps) => (
+  <div style={{color}}>
+    <Icon fa={iconType}/>
+    <span style={{marginLeft: "0.5em"}}>
+      {new Date(date).toLocaleString()}
+    </span>
+  </div>
+);
+
+const OngoingUpload = (upload: UserDatasetUpload, onClickCancel: () => void) => (
+  <div>
+    <UploadHeader color="orange" iconType="cogs" date={upload.finished || upload.started} />
+    <div> 
+      Currently uploading: <code>{upload.datasetName}</code>
+    </div>
+    <div>
+      Status: {upload.status + (upload.stepPercent ? " ... " + upload.stepPercent + "%" : "")}
+    </div>
+    {upload.status === 'awaiting-upload' && <a style={{cursor: "pointer"}} onClick={()=>onClickCancel()}>Cancel upload</a> }
```

Comment:
**nitpick:** A `button` element should be used for "Cancel upload". `a` elements should be reserved for navigation.

Reply:
Changed into a button